mantidproject/mantid
# Allow bank layout to be specified in side by side view

**Description of work.**

Update to the side by side view in the instrument viewer to allow an optional bank layout to be specified in the instrument definition file.
 
This change was requested by the SXD instrument scientists at ISIS. The SXD instrument consists of 11 square banks of detectors. We've also had a similar request from the Bilby instrument. I've made an update to the SXD IDF to use the new feature but haven't done anything to Bilby so far.

This PR also includes a change to the projection used to rotate all the banks onto the xy plane ready for the side by side view. The original code left open the possibility of a twist\yaw about the normal to the bank face during this projection. The projection now keeps the orientation of the bank about this axis the same ie if top edge of bank is horizontal it stays horizontal on the side by side view.

I have removed an unnecessary #include of the RectangularDetector.h file from ~8 files as part of this PR. While assessing the best solution for this I wanted to see where the RectangularDetector class was used and this was obscuring my view a little.

**To test:**

Load run SXD33304.nxs. Right click on the resulting workspace and select "Show Instrument" and then select Side by Side View. It should look like this - the banks may not be perfectly square depending on the window size (and also whether https://github.com/mantidproject/mantid/pull/35090 has been merged yet):

![image](https://user-images.githubusercontent.com/56295817/216916089-07342f27-e6b6-44ef-a947-5a25eb9a2a85.png)

Compare this to the same view in the original code where 1-2 banks were rotated and the bank arrangement was different.

Fixes #34560.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
